### PR TITLE
Development frontend

### PR DESCRIPTION
**Pull request by HaroonMasihDevDen**
add home, navbar and login page and login the user and redirect to home page after login and show user name and email in the navbar and show logout and do logout functionality added yet , remaining if hide user profile if user is not signed in

### CODE REVIEW

File: ecommerce_frontend/public/index.html [hunk 1]
```diff
@@ -39,5 +32,6 @@
       To begin the development, run `npm start` or `yarn start`.
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
-  </body>
-</html>
+</body>
+
+</html>
```
Comment (Kawixh) [inline ecommerce_frontend/public/index.html]: add EOL here

File: ecommerce_frontend/src/Pages/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import Cookies from 'js-cookie';
+
+export default function Home() {
+  const [userName ,setUserName]=useState(null);
+  const [userEmail ,setUserEmail]=useState(null);
+
+  const authenticate_token = async () => {
+    alert("token verification function called ");
+    try {
+      const response = await axios.get("http://localhost:3001/validate_token", {
```
Comment (Kawixh) [inline ecommerce_frontend/src/Pages/Home.jsx]: Do not hard code URLs or the base domain; use ENV for the base domain and separate files for APIs, where you can define APIs using functions that take args and return an API route for you to use.

**Review by Kawixh — CHANGES_REQUESTED** [review]
Kindly address and rectify these points prior to the next review.

File: ecommerce_frontend/src/Pages/Home.jsx (hunk 2 above)
Comment (HaroonMasihDevDen) [reply ecommerce_frontend/src/Pages/Home.jsx]: like this :

REACT_APP_API_BASE_URL=http://localhost:3000/api
,

const API_BASE_URL = process.env.REACT_APP_API_BASE_URL;
export const getApiUrl = (endpoint) => `${API_BASE_URL}${endpoint}`;
// Example API routes
export const USERS_API = getApiUrl('/users');
export const PRODUCTS_API = getApiUrl('/products');
export const ORDERS_API = getApiUrl('/orders');

// Function for dynamic endpoints
export const userDetailApi = (userId) => getApiUrl(`/users/${userId}`);
export const productDetailApi = (productId) => getApiUrl(`/products/${productId}`);

Comment (Kawixh) [reply ecommerce_frontend/src/Pages/Home.jsx]: yes indeed

**Pull request merged**